Limit names of directories and test data groups (#357)
Kattis/problem-package-format: spec/2023-07-draft.md

  ```
  i.e., they must be of length at least 2, at most 255, consist solely of lower- or uppercase letters a–z, A–Z, digits 0–9, period, dash, or underscore,
  but must not begin or end with a period, dash, or underscore.
- All directory names inside the package must match the regexp
  ```regex
  ^[a-zA-Z0-9]([a-zA-Z0-9_-]{0,253}[a-zA-Z0-9])?$
  ```

>   i.e., they must be of length at least 1, at most 255, consist solely of lower- or uppercase letters a–z, A–Z, digits 0–9, dash, or underscore,

Reviewer: ```suggestion
that is, they must be of length at least 1, at most 255, consist solely of lower- or uppercase letters a–z, A–Z, digits 0–9, dash, or underscore,
```

Test cases and groups will be used in lexicographical order on file base name.
If a specific order is desired, a numbered prefix such as `00`, `01`, `02`, `03`, and so on, can be used.
A subgroup must not have the same name as a test case in the same directory.

> E.g. if the file `data/secret/huge.in` exists then the directory `data/secret/huge/` must not, and vice versa.

Reviewer: ```suggestion
For example, if the file `data/secret/huge.in` exists then the directory `data/secret/huge/` must not, and vice versa.
```
